sections/introduction.tex

```diff
@@ -1,69 +1,109 @@
-cover both conceptual material and more ``hands-on'' tasks, because the
+work. It is also particularly useful for topics that jointly
+covers both conceptual material and more ``hands-on'' tasks, because the
 condensed time leaves more room for experimentation, discussion, and active
 learning \citep{Bransford2000-lu, Papert1980-fh}. In addition, due to the
 interactive setup students are more likely to work collaboratively and learn
-from each other. Finally, the opt-in nature of these
-courses often ensures that students are more motivated to learn, and their
-proximity both to one another as well as to the instructors makes for a good
-learning environment.
+from each other.
 
 However, there are still many challenges associated with this class structure.
 Students interact with material on their own, and their learning experience is
 heavily dependent on the ability of each student to get started in the first
 place. Because the instructional materials are developed on instructor
 computers, differences between instructors' and students' computers (e.g., memory
-available, operating system, etc.) can have hard-to-predict consequences, such
-as slow execution, or even complete failure to execute (e.g. because of the
-missing software dependencies that are not installed in students' computers).
-As these types of courses are often relatively short, small delays can result
-in a significant loss of instructional time.
+available, operating system, etc.) can have hard-to-predict consequences such
+as slow execution (e.g., due to sub-par student hardware) or a failure to
+execute code (e.g. because of missing software dependencies on
+students' computers). As these types of courses are often relatively short,
+small delays result in a significant loss of instructional time.
 
-One solution to mitigate many of these challenges is to offload the issue of
+A solution to mitigate many of these challenges is to offload the issue of
 student-specific hardware onto a shared cloud computing platform. This approach
 standardizes the experience of each student by allowing them access to a single
-online resource for the duration of the class.
-
-This article covers our recent experience implementing this approach, using
-advanced cyberinfrastructure to teach a day-long bootcamp in machine learning at
-the University of California in San Francisco. This paper describes the
-technical tool-chain and processes that we used in designing this course. This
-unique design includes hosting the course materials online and
-making them available for students to run through a browser on their own laptops,
-with no additional installation required. We will discuss the challenges in
-implementing this course setup effectively, and discuss its merits and
-drawbacks.
+online resource for the duration of the class, without requiring any new
+software or data to be downloaded on student computers. Next we describe
```

Comment:
Students can work on the class material simultaneously with the teacher and can try different solutions on their cloud copy of the material. This allows in comparison to traditional bootcamp experience a more interactive teaching and learning environment.

<<Trying to address 1st rev. 1st point a little further>>